nexus/src/app/support_bundles.rs

```diff
@@ -123,6 +123,7 @@ impl super::Nexus {
                         &ZpoolUuid::from(bundle.zpool_id),
                         &DatasetUuid::from(bundle.dataset_id),
                         &SupportBundleUuid::from(bundle.id),
+                        range,
```

Comment:
This looks like a minor change, but this part is actually critical: for all these APIs, where Nexus was proxying requests to the sled agent over progenitor APIs, it was previously unable to forward range request headers.

Now, with the new progenitor work, it can do so, for all bundle-related APIs.

(This is wiring up the same thing we did already in https://github.com/oxidecomputer/omicron/pull/8183 , but for other more granular bundle access)